lib/eight_ball/marshallers/json.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require 'awrence'
+require 'json'
+require 'plissken'
+
+# A JSON marshaller can convert back and forth between JSON and a list of {EightBall::Feature Features}
+# The JSON produced will be pretty-printed, as it is assumed the output will be written to a file.
+#
+# When converting from JSON, the top-level JSON element must be an array and
+# its keys must use camel-case; this will be converted to snake-case by EightBall
+# in order to adhere to both JSON and Ruby standards.
+#
+# Below are some examples of valid JSON:
+#
+# @example A single {EightBall::Feature} is enabled for accounts 1-5 as well as region Europe
+#   [{
+#     "name": "Feature1",
+#     "enabledFor": [{
+#       "type": "range",
+#       "parameter": "accountId",
+#       "min": 1,
+#       "max": 5
+#     }, {
+#       "type": "list",
+#       "parameter": "regionName",
+#       "values": ["Europe"]
+#     }]
+#   }]
+#
+# @example A single {EightBall::Feature} is disabled completely using the {EightBall::Conditions::Always Always} condition
+#   [{
+#     "name": "Feature1",
+#     "disabledFor": [{
+#       "type": "always"
+#     }]
+#   }]
+module EightBall::Marshallers
+  class Json
+    # Convert the given {EightBall::Feature Features} into a JSON array.
+    #
+    # @param [Array<EightBall::Feature>] features The {EightBall::Feature Features} to convert.
+    # @return [String] The resulting JSON string.
+    #
+    # @example
+    #   json_string = <Read from somewhere>
+    #
+    #   marshaller = EightBall::Marshallers::Json.new
+    #   marshaller.unmarshall json_string => [Features]
+    def marshall(features)
+      JSON.pretty_generate(features.map { |feature| feature_to_hash(feature).to_camelback_keys })
```

Comment:
Why does it have to be pretty?

Reply:
It doesn't. It's a choice I made because I'm thinking that 99% of the time the reason to marshall something to JSON would be to put it in a file. Files are nice when they're readable. 

I'm open to having my mind changed.

Reply:
I was hoping that we didn't have to read the file. It's either accessed programmatically or by a developer who's text editor would take care of it.

I think we're talking about saving pennies in storage though, haha. Thinking about that though, I'd be curious about the throughput rates for the file sizes.

Reply:
I've removed the pretty-printing after a couple more tests. It ended up doubling the output size and it is easy to pretty-print after the fact it so desired.